crates/sequencer/src/commitment_controller.rs

```diff
@@ -46,15 +46,31 @@ pub fn get_commitment_info(
     // if there is a height then start from height + 1 and go to prev_l1_height
     let (l2_range_to_submit, l1_height_range) = match last_commitment_l1_height {
         Some(last_commitment_l1_height) => {
-            let l1_height_range = (last_commitment_l1_height.0 + 1, prev_l1_height);
+            let l1_start = last_commitment_l1_height.0;
+            let mut l1_end = l1_start;
+            let mut l2_length = 0;
+            // Take while sum of l2 ranges <= min_soft_confirmations_per_commitment
+            while l2_length <= min_soft_confirmations_per_commitment
```

Comment:
l2_length < min_soft_confirmations_per_commitment


we can commit on l2_length = min_soft_confirmations_per_commitment

Reply:
I dont understand